src/index.js

```diff
@@ -0,0 +1,132 @@
+const state = {
+  temp: 68
+};
+
+let isFahrenheit = true;
+
+const vibe = {
+  sheet: 2
+}
+
+let isReset = false;
+
+const changeTemp = (direction) => {
+  if (direction === 'up') {
+    state.temp += 1;
+  } else if (direction === 'down') {
+    state.temp -= 1;
+  } else if (direction === null) {
+    pass
+  }
+
+  const temperature = document.querySelector("#temperature");
+  if (isFahrenheit) {
+    temperature.textContent = `${state.temp}℉`;
+  } else {
+    temperature.textContent = `${(state.temp -32) * 5/9}℃`;
+  }
+  changeTempFontColor();
+}
+
+
+const swapStyleSheet = (event) => {
+
+  if (vibe.sheet === 1 || vibe.sheet === 2) {
+      vibe.sheet += 1;
+  } else if (vibe.sheet === 3) {
+      vibe.sheet = 1;
+  }
+
+  document.getElementById("vibe").setAttribute("href", `styles/vibe${vibe.sheet}.css`)
+}
+
+const changeTempFontColor = () => {
+  if (state.temp <= 49) {
+    color = "#3F50B5";
+    landscape = "🤧⛄️🌲⛄️🛷🌲⛷🗻🌲⛸🏂⛄️🍂❄️";
+  } else if (state.temp >= 50 && state.temp<= 59) {
+    color = "#00BCD4";
+    landscape = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+  } else if (state.temp >= 60 && state.temp  <= 69) {
+    color = "#FEC106";
+    landscape = "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃";
+  } else if (state.temp >= 70 && state.temp  <= 79) {
+    color = "#FF9802";
+    landscape = "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷";
+  } else {
+    color = "#F44336";
+    landscape = "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂";
+  }
```

Comment:
Consider the above logic and how similar all the branches are. We look for whether the temperature is within a range, and pick a color and landscape accordingly. What if we had a list of records that we could iterate through to fin d the values. As noted above, we really only need to store the top of each range, so we could set up something like
```js
  const tempRecords = [
    { upperBound: 49, color: "#3F50B5", landscape: "🤧⛄️🌲⛄️🛷🌲⛷🗻🌲⛸🏂⛄️🍂❄️"},
    { upperBound: 59, color: "other color", landscape: "other landscape"},
    ...
  ];
```

For the `else` case, we might set the `upperBound` to a very high value, or to `null` and write a special case to check it. But for everything else, we could simply iterate through the list and find the first record that has an upper bound _lower_ than our temperature, the use the prior record (the last record that had an upper bound _higher_ than our temperature) as the source of picking the color and landscape!